source/common/upstream/upstream_impl.h

```diff
@@ -101,6 +102,80 @@ class UpstreamLocalAddressSelectorImpl : public UpstreamLocalAddressSelector {
   std::vector<UpstreamLocalAddress> upstream_local_addresses_;
 };
 
+/**
+ * Class for LBPolicies
+ * Uses a absl::variant to store pointers for the LBPolicy
+ */
+class LBPolicyConfig {
+public:
+  LBPolicyConfig(const envoy::config::cluster::v3::Cluster& config);
+
+  OptRef<const envoy::config::cluster::v3::Cluster::RoundRobinLbConfig> lbRoundRobinConfig() const {
+    if (auto lb_ptr = absl::get_if<
```

Comment:
nit: using `std::holds_alternative` might make the code more readable. 

Also, it's probably better to have a templated helper method to avoid the repeated code.

Reply:
Done